File: backend/BlockChain.Domain/BlockChain/Usuario.cs
using BlockChain.Cross.Entity;
using BlockChain.Cross.Utils;
using BlockChain.Domain.BlockChain.Rules;
using BlockChain.Domain.BlockChain.ValueObject;
using FluentValidation;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlockChain.Domain.BlockChain
{
    public class Usuario : Entity<Guid>
    {
        public string Nome { get; set; }
        public Email Email { get; set; }
        public Password Password { get; set; }
        public IList<Carteira> Carteiras { get; set; }

        public void SetPassword()
        {
            this.Password.Valor = SecurityUtils.HashSHA1(this.Password.Valor);
        }

        public void Validate() =>
            new UsuarioValidator().ValidateAndThrow(this);


    }
}

File: frontend/src/app/model/Usuario.ts
import { BaseEntity } from './BaseEntity';
import { Carteira } from './Carteira';

/*import { TipocomandOperacao } from './TipocomandOperacao';
import { TipoMotivorestricao } from './TipoMotivorestricao';
import { ComandOperacao } from './ComandOperacao';*/


export class Usuario extends BaseEntity<number> {


    public Nome?: string;

    public Email?: string;
    
    public Password?: string;

    public CarteirasLista?: Carteira;
    

}